tool/formula/src/main/java/org/openscience/cdk/tools/manipulator/MolecularFormulaManipulator.java

```diff
@@ -209,14 +209,31 @@ public static String getString(IMolecularFormula formula, String[] orderElements
             if (!elemSet.contains(symbol)) elemSet.add(symbol);
         }
 
-        for (String elem : elemSet) {
-            int count = 0;
-            for (IIsotope isotope : formula.isotopes()) {
-                if (isotope.getSymbol().equals(elem)) count += formula.getIsotopeCount(isotope);
+        if (!setMassNumber) {
+            for (String elem : elemSet) {
+                int count = 0;
+                for (IIsotope isotope : formula.isotopes()) {
+                    if (isotope.getSymbol().equals(elem)) count += formula.getIsotopeCount(isotope);
+                }
+                stringMF.append(elem);
+                if (!(count == 1 && !setOne)) stringMF.append(count);
+            }
+        } else {
+            for (IIsotope isotope : isotopesList) {
+                int count = formula.getIsotopeCount(isotope);
+                try {
+                    IIsotope major = Isotopes.getInstance().getMajorIsotope(isotope.getSymbol());
+                    if (Objects.equals(isotope.getMassNumber(), major.getMassNumber()))
+                        stringMF.append(isotope.getSymbol());
+                    else
+                        stringMF.append("["+isotope.getMassNumber()+isotope.getSymbol()+"]");
```

Comment:
should be multiple appends: append().append() IDE is probably providing hint to auto-update on this.